Packs/Shodan/Integrations/Shodan_v2/Shodan_v2.py

```diff
@@ -45,7 +45,8 @@ def http_request(method, uri, params=None, data=None, headers=None):
                            data=data,
                            headers=headers,
                            verify=USE_SSL)
-
+    if res.status_code == 404:
+        return {}
     if res.status_code != 200:
         error_msg = f'Error in API call {url} [{res.status_code}] - {res.reason}'
```

Comment:
Why does this is not thrown?

Reply:
there isnt even main function here and dont want to change the whole integration

Reply:
I think it probably makes more sense to `raise_for_status()` here and then use a `try-except` block below where if it's a 404 you can say no details were found instead of raising an error.

Reply:
i dont think it really matters at this point, it does the same thing

Reply:
It moves all of the handlings of custom logic to the calling function instead of performing custom behavior in the function that makes the request itself.

Reply:
i prefer not to change this as `raise_for_status` accepts all status codes that are not in the range of 400-600, and some apis return 2xx and 3xx as incomplete results and stuff like that - we might break something here